mico-admin/src/app/app-detail/app-detail.component.ts

```diff
@@ -16,12 +16,35 @@ export class AppDetailComponent implements OnInit {
         private route: ActivatedRoute,
     ) { }
 
-    @Input() application: ApiObject;
+    application: ApiObject;
+    selectedVersion;
 
     ngOnInit() {
         const id = +this.route.snapshot.paramMap.get('id');
-        this.apiService.getApplicationById(id)
-            .subscribe(app => this.application = app);
+        // getServiceVersions works also for applications
```

Comment:
ngOnInit is insufficient if only the rout param changes as the router will not recreate the component.
if possible use an `@Input` with ngOnChanges or properly subscribe to route param changes.
If not fixed now add a TODO comment